spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/dynamodb/DynamoDbAutoConfiguration.java

```diff
@@ -46,19 +53,50 @@
 @AutoConfigureAfter({ CredentialsProviderAutoConfiguration.class, RegionProviderAutoConfiguration.class })
 @ConditionalOnProperty(name = "spring.cloud.aws.dynamodb.enabled", havingValue = "true", matchIfMissing = true)
 public class DynamoDbAutoConfiguration {
+	@ConditionalOnProperty(name = "spring.cloud.aws.dynamodb.dax.url")
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(name = "software.amazon.dax.ClusterDaxClient")
+	static class DaxDynamoDbClient {
 
-	private final DynamoDbProperties properties;
+		@ConditionalOnMissingBean
+		@Bean
+		public DynamoDbClient dynamoDbClient(DynamoDbProperties properties, AwsCredentialsProvider credentialsProvider,
+				AwsRegionProvider regionProvider, ObjectProvider<List<MetricPublisher>> metricsPublishers)
```

Comment:
Why do we have support here for metrics publisher?

```
ObjectProvider<List<MetricPublisher>> metricsPublishers
```

I get that it's good to have but i guess we should add it at the same time to all clients? Or is there something specific to DAX?

If we want to have here support for metrics publisher, shouldn't it be `ObjectProvider<MetricPublisher>`?

Reply:
Metric publishers are something that is unique to DAX. 
`Configuration` that configures a client accepts List of MetricPublishers. 
`private List<MetricPublisher> metricPublishers;`

Since it can have more MetricPublishers I think we need to support it as a List.

Reply:
Each client can have metrics publishers - this is not unique to DAX.

Also even if it is a list, you can (and i think should) use `ObjectProvider<MetricsPublisher>` and then call `stream()` on it to get all autowired elements.

Since we do not support autoconfiguring metrics publisher anywhere else I believe we should drop it from here too.